### PR TITLE
Help buttons

### PR DESCRIPTION
**Pull request by jwaltt55**

### CODE REVIEW

**Review by dyl10s — CHANGES_REQUESTED** [review]
It looks a bit weird on mobile, maybe if it covered the whole page and had some margin it would be better?

image

Also instead of repeating this html and css on the different components that needed it what about breaking it into its own component so you can just write like

```
<help-icon>
    This is the help text here <br/>
    this would eliminate repeated code  
</help-icon>
```

File: Frontend/src/app/views/dashboard/report-details/report-details.component.html [hunk 1]
```diff
@@ -77,5 +77,24 @@ <h3>Detailed Time Report</h3>
...
+    <nb-icon icon="question-mark-circle"></nb-icon>
+  </div>
+
+  <nb-layout id="popoverStyle">
+    <ng-template #popoverTemplate>
+      <p>
+        This is the Detailed Time Report page, where you can view a list of time entries for a specific project.<br><br>
+        <u>Adjust the Date Range</u> for the displayed results by using the date range selector at the top of the screen.<br>
+        The dropdown menu allows you to choose a specific time interval. Clicking the arrow buttons will change the dates by that interval.<br>
+        Clicking on the date boxes will also let you change the date range by typing in a new date or using the calander menu.<br><br>
```
Comment (dyl10s) [inline Frontend/src/app/views/dashboard/report-details/report-details.component.html]: Typo `calander`

File: Frontend/src/app/views/dashboard/report/report.component.html [hunk 2]
```diff
@@ -51,4 +51,24 @@ <h4>Projects</h4>
...
+        <nb-icon icon="question-mark-circle"></nb-icon>
+      </div>
+    
+      <nb-layout id="popoverStyle">
+        <ng-template #popoverTemplate>
+          <p>
+            This is the main Reports page, where you can view the amount of time spent on each of your projects by all team members.<br><br>
+            <u>Adjust the Date Range</u> for the displayed results by using the date range selector at the top of the screen.<br>
+            The dropdown menu allows you to choose a specific time interval. Clicking the arrow buttons will change the dates by that interval.<br>
+            Clicking on the date boxes will also let you change the date range by typing in a new date or using the calander menu.<br><br>
```
Comment (dyl10s) [inline Frontend/src/app/views/dashboard/report/report.component.html]: typo `calander`

File: Frontend/src/app/views/dashboard/project-details/project-details.component.scss [hunk 3]
```diff
@@ -66,5 +66,4 @@ table {
       padding: 0 !important;
       margin: 0 !important;
     }
-}
-
+}
```
Comment (dyl10s) [inline Frontend/src/app/views/dashboard/project-details/project-details.component.scss]: Remove this change

File: Frontend/src/app/views/dashboard/projects/projects.component.ts [hunk 4]
```diff
@@ -8,6 +8,7 @@ import { CustomFilterService } from 'src/app/core/services/customFilterService.s
 import { CustomTreeBuilder } from 'src/app/core/services/customTreeBuilder.service';
 import { ProjectService } from 'src/app/core/services/project.service';
 import { CreateProjectComponent } from 'src/app/shared/components/create-project/create-project.component';
+import { NbPopoverDirective } from '@nebular/theme';
```
Comment (dyl10s) [inline Frontend/src/app/views/dashboard/projects/projects.component.ts]: Do you still need this import?

File: Frontend/src/app/views/dashboard/report-details/report-details.component.scss [hunk 5]
```diff
@@ -70,5 +70,4 @@
       margin-left: auto;
     }
   }
-}
-
+}
```
Comment (dyl10s) [inline Frontend/src/app/views/dashboard/report-details/report-details.component.scss]: Remove this change

File: Frontend/src/app/views/dashboard/report/report.component.scss [hunk 6]
```diff
@@ -46,6 +46,4 @@
     padding: 0 !important;
     margin: 0 !important;
     }
-}
-
-
+}
```
Comment (dyl10s) [inline Frontend/src/app/views/dashboard/report/report.component.scss]: Remove this change

File: Frontend/src/app/views/dashboard/time/time.component.scss [hunk 7]
```diff
@@ -228,4 +228,4 @@
     padding: 0 !important;
     margin: 0 !important;
   }
-}
+}
```
Comment (dyl10s) [inline Frontend/src/app/views/dashboard/time/time.component.scss]: Remove this change

**Review by dyl10s — CHANGES_REQUESTED** [review]

**Review by dyl10s — APPROVED** [review]

**Review by tonae-patterson — APPROVED** [review]

**Pull request merged**